python/pfs/drp/stella/lsf.py

```diff
@@ -691,3 +692,100 @@ def computeKernel(self, center):
     def __reduce__(self):
         """Pickling"""
         return self.__class__, (self.kernel, self.length, self.interpolator)
+
+
+class ExtractionLsf(Lsf):
+    """LSF from extracting a PSF
+
+    Parameters
+    ----------
+    psf : `pfs.drp.stella.SpectralPsf`
+        Point-spread function for spectral data.
+    fiberTrace : `pfs.drp.stella.FiberTrace`
+        Fiber profile.
+    length : `int`
+        Array length.
+    """
+    def __init__(self, psf, fiberTrace, length):
+        self.psf = psf
+        self.fiberTrace = fiberTrace
+        self.fiberId = fiberTrace.fiberId
+        super().__init__(length)
+
+    def computeKernel(self, yy):
+        """Return a kernel for the nominated position
+
+        Besides the difference in return types from the ``computeArray``
+        method (this returns a `Kernel1D`, while ``computeArray`` returns a
+        an array), this method provides a model for the LSF centered at zero.
+
+        This implementation realises the PSF and extracts it with the
+        fiberTrace. The difference from ``computeArray` is that this method
+        puts the center on the middle of a pixel.
+
+        Parameters
+        ----------
+        center : `float`, pixels
+            Position at which to evaluate the LSF. Not that it matters, as it's
+            the same everywhere.
+
+        Returns
+        -------
+        kernel : `Kernel1D`
+            Kernel with a model of the LSF at the nominated position.
+        """
+        detMap = self.psf.getDetectorMap()
+        xx = detMap.getXCenter(self.fiberId, yy)
+        psfImage = self.psf.computeKernelImage(Point2D(xx, yy)).convertF()
+        xy0 = psfImage.getXY0()
+        psfImage.setXY0(xy0 + Extent2I(int(xx + 0.5), int(yy + 0.5)))  # So extraction works properly
+        traces = FiberTraceSet(1)
+        traces.add(self.fiberTrace)
+        spectra = traces.extractSpectra(MaskedImageF(psfImage))
+        assert len(spectra) == 1
+        return Kernel1D(spectra[0].spectrum, -xy0.getY())
+
+    def computeArray(self, yy):
+        """Return an array with the LSF inserted at the nominated position
+
+        Besides the difference in return types from the ``computeKernel``
+        method (this returns an array, while ``computeKernel`` returns a
+        `Kernel1D`), this method allows positioning of the LSF at sub-pixel
+        positions. This is the method you want to use if you want a model for
+        a particular line in a spectrum, e.g., to measure the flux and/or
+        subtract the line.
+
+        This implementation realises the PSF and extracts it with the
+        fiberTrace. The difference from ``computeArray` is that this method
```

Comment:
This method is also called ``computeArray``. Are you referring to a different method?